be/src/exec/olap_table_info.h

```diff
@@ -105,12 +105,25 @@ struct OlapTablePartition {
 class OlapTablePartKeyComparator {
 public:
     OlapTablePartKeyComparator(SlotDescriptor* slot_desc) : _slot_desc(slot_desc) { }
+    // return true if lhs < rhs
+    // 'nullptr' is max value, but 'null' is min value
     bool operator()(const Tuple* lhs, const Tuple* rhs) const {
         if (lhs == nullptr) {
             return false;
         } else if (rhs == nullptr) {
             return true;
         }
+
+        bool lhs_null = lhs->is_null(_slot_desc->null_indicator_offset());
+        bool rhs_null = rhs->is_null(_slot_desc->null_indicator_offset());
+        if (lhs_null && rhs_null) {
+            return false;
+        } else if (lhs_null) {
+            return true;
+        } else if (rhs_null) {
+            return false;
+        }
+
```

Comment:
It exists some redundant if. you can change to 

if (rhs_null) {
    return false;
} else if (lhs_null) {
   return true;
}
 
or 

if (lhs_null || rhs_null) {
    return rhs_null? false : true;
}